response.go

```diff
@@ -88,12 +86,8 @@ func (r *Response) fixCharset(detectCharset bool, defaultEncoding string) error
 		if !detectCharset {
 			return nil
 		}
-		d := chardet.NewTextDetector()
-		r, err := d.DetectBest(r.Body)
-		if err != nil {
-			return err
-		}
-		contentType = "text/plain; charset=" + r.Charset
+		_, nameOfEncoding, _ := charset.DetermineEncoding(r.Body, contentType) //name of charset/encoding
+		contentType = "text/plain; charset=" + nameOfEncoding
```

Comment:
`charset.DetermineEncoding` returns the detected encoding, so I think this entire function can be greatly simplified and replaced with rougly this:

```go
enc, _, certain := charset.DetermineEncoding(r.Body, contentType)
if !certain && !detectCharset {
	return nil
}
var err error
r.Body, err = ioutil.ReadAll(enc.NewDecoder().Reader(bytes.NewReader(r.Body)))
return err
```